### PR TITLE
[GNA] Plugin transition to the library v3.0 Master

### PR DESCRIPTION
**Pull request by andreysapozhn**
### Details:
 - Changes from [PR-7047](https://github.com/openvinotoolkit/openvino/pull/7047) and [PR-7241](https://github.com/openvinotoolkit/openvino/pull/7241) on `releases/2021/4` branch for transition to the GNA library v3.0

### Tickets:
 - *55660*

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 1]
```diff
@@ -170,21 +170,17 @@ void GNADeviceHelper::releaseModel(const uint32_t model_id) {
 
 bool GNADeviceHelper::enforceLegacyCnnNeeded() const {
     const auto compileTargetDevice = getTargetDevice(false);
-    return isGnaLibVersion2_1 && isUpTo20HwGnaDevice(compileTargetDevice);
+    return (isGnaLibVersion3_0 || isGnaLibVersion2_1) && isUpTo20HwGnaDevice(compileTargetDevice);
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/gna_device.cpp:173]: Should 2.1 stay here and below or do we need only 3.0?

File: inference-engine/src/gna_plugin/gna_device.hpp [hunk 2]
```diff
@@ -98,6 +105,9 @@ class GNADeviceHelper {
         if (gnaLibVersion.rfind("2.1", 0) == 0) {
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/gna_device.hpp:105]: Same here - do we need 2.1?

File: inference-engine/src/gna_plugin/gna_device.cpp (hunk 1 above)
Comment (andreysapozhn) [reply inference-engine/src/gna_plugin/gna_device.cpp:173]: @sirzabek, @kbruniec asked to save this option

**Review by sirzabek — APPROVED** [review]

File: inference-engine/src/gna_plugin/transformations/decompose_2d_convolution.cpp [hunk 3]
```diff
@@ -349,7 +370,7 @@ static std::shared_ptr<ngraph::Node> CreateDecomposedConv(const GraphData& graph
             // We need to calculate some parameters in case horizontal stride > 1 is used, because if we use the ones available from the original convolution
             // we won't take into account the fact horizontal strides will be supported by the newly created 1D convolution, and not by decomposition
             size_t filter_dilation_width = conv_data.filter_width > 1 ? conv_data.filter_dilation_width : 1;
-            size_t output_width = (conv_data.input_width - (conv_data.filter_width + filter_dilation_width - 2));
+            size_t output_width = (conv_data.input_width - (filter_dilation_width * (conv_data.filter_width - 1)));
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/transformations/decompose_2d_convolution.cpp:373]: Please clarify two things:
1. Is `filter_dilation_width != 1` really supported here?
2. Is only convolution horizontal stride == 1 supported ?

**Review by kbruniec — APPROVED** [review]
LGTM

File: inference-engine/src/gna_plugin/transformations/decompose_2d_convolution.cpp (hunk 3 above)
Comment (sirzabek) [reply inference-engine/src/gna_plugin/transformations/decompose_2d_convolution.cpp:373]: 1. Yes, there are such cases in tests. Why?
2. No, please see comments. above the code. There are test cases with horizontal stride > 1. Again, why?

Comment (kbruniec) [reply inference-engine/src/gna_plugin/transformations/decompose_2d_convolution.cpp:373]: Thanks for explanations.

**Pull request merged**